src/components/Layout/MDXWrapper.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { PropsWithChildren } from 'react';
+import { navigate, PageProps } from 'gatsby';
+import CodeSnippet, { CodeSnippetProps } from '@ably/ui/core/CodeSnippet';
+
+import '../../styles/global.css';
+import PageTitle from '../PageTitle';
+import { MarkdownProvider } from '../Markdown';
+import Article from '../Article';
+import If from './mdx/If';
+import { useCopyableHeaders } from './mdx/headers';
+import { useLayoutContext } from 'src/contexts/layout-context';
+import cn from '@ably/ui/core/utils/cn';
+import Aside from '../blocks/dividers/Aside';
+import { HtmlComponentPropsData } from '../html-component-props';
+import { PageContextType } from './Layout';
+
+type MDXWrapperProps = PageProps<unknown, PageContextType>;
+
+const MDXWrapper: React.FC<MDXWrapperProps> = ({ children, pageContext }) => {
+  const { frontmatter } = pageContext;
+  const title = frontmatter?.title;
+  const { activePage, setLanguage } = useLayoutContext();
+
+  // Use the copyable headers hook
+  useCopyableHeaders();
+
+  const WrappedCodeSnippet: React.FC<CodeSnippetProps> = (props) => {
+    return (
+      <CodeSnippet
+        lang={activePage.language}
+        onChange={(lang) => {
+          setLanguage(lang);
+          window.history.replaceState({}, '', `${location.pathname}?lang=${lang}`);
+        }}
+        className={cn(props.className, 'mb-20')}
+        data-languages="javascript-react"
+        {...props}
+      />
+    );
+  };
+
+  const WrappedAside = (props: PropsWithChildren<{ 'data-type': string }>) => {
+    return (
+      <Aside
+        attribs={{ 'data-type': props['data-type'] }}
+        data={(<>{props.children}</>) as unknown as HtmlComponentPropsData}
+      />
+    );
+  };
+
+  // Table components with responsive styling
+  const Table = (props: React.HTMLAttributes<HTMLTableElement>) => (
```

Comment:
Very minimal table styling. This is basically here to expose an entry point for better styling later.